src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -48,12 +48,16 @@ BoundExpression TranslatePattern(BoundExpression input, BoundPattern pattern)
                         input = _factory.Local(temp);
                         for (int i = 0; i < pat.Subpatterns.Length; i++)
                         {
-                            var subProperty = pat.Subpatterns[i].Property;
+                            var subProperty = (pat.Subpatterns[i].Member as BoundPropertyPatternMember)?.MemberSymbol;
                             var subPattern = pat.Subpatterns[i].Pattern;
                             var subExpression =
-                                subProperty.Kind == SymbolKind.Field
-                                    ? (BoundExpression)_factory.Field(input, (FieldSymbol)subProperty)
-                                    : _factory.Call(input, ((PropertySymbol)subProperty).GetMethod);
+                                subProperty?.Kind == SymbolKind.Field ?
+                                    _factory.Field(input, (FieldSymbol)subProperty) :
+                                subProperty?.Kind == SymbolKind.Property ?
+                                    _factory.Call(input, ((PropertySymbol)subProperty).GetMethod) :
```

Comment:
What happened with all the arguments, etc.? It feels like we need a test case for a scenario when arguments are used.